misc/formatter/modules/formatter-core/src/main/java/org/ballerinalang/formatter/core/FormattingTreeModifier.java

```diff
@@ -596,19 +630,30 @@ public ImportPrefixNode transform(ImportPrefixNode importPrefixNode) {
     public ServiceDeclarationNode transform(ServiceDeclarationNode serviceDeclarationNode) {
         MetadataNode metadata = formatNode(serviceDeclarationNode.metadata().orElse(null), 0, 1);
         Token serviceKeyword = formatToken(serviceDeclarationNode.serviceKeyword(), 1, 0);
-        //IdentifierToken serviceName = formatToken(serviceDeclarationNode.serviceName().orElse(null), 1, 0);
+        TypeDescriptorNode typeDescriptor = formatNode(serviceDeclarationNode.typeDescriptor().orElse(null), 1, 0);
+        NodeList<Node> absoluteResourcePath = formatNodeList(serviceDeclarationNode.absoluteResourcePath(), 0, 0, 1, 0);
         Token onKeyword = formatToken(serviceDeclarationNode.onKeyword(), 1, 0);
         SeparatedNodeList<ExpressionNode> expressions =
                 formatSeparatedNodeList(serviceDeclarationNode.expressions(), 0, 0, 1, 0);
-        //Node serviceBody = formatNode(serviceDeclarationNode.serviceBody(), env.trailingWS, env.trailingNL);
+        Token openBrace = formatToken(serviceDeclarationNode.openBraceToken(), 0, 1);
+        indent(); // increase the indentation of the following statements.
+        NodeList<Node> members = formatNodeList(serviceDeclarationNode.members(), 0, 1, 0, 1, true);
+        unindent(); // reset the indentation.
+        if (serviceDeclarationNode.members().isEmpty()) {
+            env.preserveNewlines = true;
+        }
```

Comment:
Do we need to check the empty-ness?

Reply:
Since a similar behavior is implemented here in `FunctionBodyBlockNode`, the newline preservation was added here as well. 

https://github.com/ballerina-platform/ballerina-lang/blob/7289b5b33d06787e4ebfa7159d6eab19e5545f80/misc/formatter/modules/formatter-core/src/main/java/org/ballerinalang/formatter/core/FormattingTreeModifier.java#L361

Otherwise, the following parser-tests seems to be failing as well. 

https://github.com/ballerina-platform/ballerina-lang/blame/master/compiler/ballerina-parser/src/test/resources/declarations/service-decl/service_decl_source_18.bal 

https://github.com/ballerina-platform/ballerina-lang/blame/master/compiler/ballerina-parser/src/test/resources/declarations/service-decl/service_decl_source_19.bal

Reply:
Preserving the newline is fine. But do we need to check for the emtpy-ness?
As in, are we not preserving the newlines if the list is not empty?

Reply:
Fixed in `FunctionBodyBlockNode` and `ServiceDeclarationNode`.